lib/ukrust/target.json

```diff
@@ -0,0 +1,37 @@
+
+{
+  "arch": "x86_64",
+  "code-model": "kernel",
+  "cpu": "x86-64",
+  "data-layout": "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128",
+  "disable-redzone": true,
+  "emit-debug-gdb-scripts": false,
+  "env": "gnu",
+  "features": "-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-3dnow,-3dnowa,-avx,-avx2,+soft-float",
```

Comment:
I think the CPU features should be selected depending on the selcted target CPU. We risk that we compile code always with AVX which would us prevent to run on non-AVX CPUs - is this true?

Reply:
btw, this file could be generated with `M4` or using the gcc preprocessor only

Reply:
Actually, we instruct rustc not to use any SIMD related instructions hence the `-`, we only enable soft floats. I'm not sure if, at least right now, we want to generate this file. I'd leave it for a future pull request If the needs arise. Most project use it as this.